clustering/clustering.py

```diff
@@ -0,0 +1,446 @@
+
+    def cluster_graph_overlap(sefl, graph, k=5):
+        """Applies overlapping community detection in the given graph.
+
+        Uses the percolation method for finding the k-clique communities in the given graph.
+        This method returns 4 elements to follow the same structure as cluster_graph().
+        However the modularity of an overlapping partition cannot be defined, we return None instead.
+
+        Args:
+            graph (nx.Graph): NetworkX graph representation of sounds.
+
+        Returns:
+            Tuple(Dict{Int: Int}, int, List[List[Int]], None): 4-element tuple containing the clustering classes for each sound 
+                {<sound_id>: <class_idx>}, the number of communities (clusters), the sound ids in the communities and
+                None.
+        """ 
+        communities = [list(community) for community in k_clique_communities(graph, k)]
+        # communities = [list(community) for community in greedy_modularity_communities(graph)]
+        greedy_modularity_communities
+        num_communities = len(communities)
+        classes = {sound_id: cluster_id for cluster_id, cluster in enumerate(communities) for sound_id in cluster}
+
+        return  classes, num_communities, communities, None
+
+    def remove_lowest_quality_cluster(self, graph, classes, communities, ratio_intra_community_edges):
+        """Removes the lowest quality cluster in the given graph.
+
+        Discards the cluster that has the lowest ratio of the number of intra-community edges. Removes the related values 
+        of the removed cluster from the given clustering information.
+
+        Args:
+            graph (nx.Graph): NetworkX graph representation of sounds.
+            classes (Dict{Int: Int}): clustering classes for each sound {<sound_id>: <class_idx>}.
+            communities (List[List[Int]]): List storing Lists containing the Sound ids that are in each community (cluster).
+            ratio_intra_community_edges (List[Float]): intra-community edges ratio.
+
+        Returns:
+            Tuple(nx.Graph, Dict{Int: Int}, List[List[Int]], List[float]): 4-element tuple containing the graph representation 
+            of the sounds, the clustering classes for each sound {<sound_id>: <class_idx>}, the sound ids in the communities 
+            and the ratio of intra-community edges in each cluster.
+        """
+        # if two clusters or less, we do not remove any
+        if len(communities) <= 2:
+            return graph, classes, communities, ratio_intra_community_edges
+        min_ratio_idx = np.argmin(ratio_intra_community_edges)
+        sounds_to_remove = communities[min_ratio_idx]
+        graph.remove_nodes_from(sounds_to_remove)
+        del communities[min_ratio_idx]
+        for snd in sounds_to_remove:
+            del classes[snd]
+        del ratio_intra_community_edges[min_ratio_idx]
+        for idx in range(min_ratio_idx, max(classes.values())):
```

Comment:
at https://github.com/MTG/freesound/pull/1389/files#diff-40410ace29c087b6ebccb3110c2a4c6cR312 you do `range(max(classes.values())+1)`. Is this one supposed to be different? I wonder if you could use a separate method with an explicit name in order to make it clear what this is doing